### DISCUSSION TITLE
Are commands run lazily?

### DISCUSSION THREAD

**Discussion by LanceEa** [redis/go-redis#2308] [Q&A] [answered]
In the docs [executing-commands](https://redis.uptrace.dev/guide/go-redis.html#executing-commands) it says the command can be stored and then the val/error accessed later. Does this mean the command is not executed until the val/error are asked for or does the command get executed right away?

For example, if try storing the command like such does it run when I call `Get` or when I try to access the `Result`?
```golang
var cmd *redis.StatusCmd

if foo == true {
    cmd = rdb.Get(ctx, "foo")
} else {
    cmd = rdb.Get(ctx, "bar")
}

val, err := cmd.Result()
```

Thanks for the help!

**Comment by vmihailenco** [comment, marked as answer]
Commands run synchronously when you execute `rdb.Get(ctx, "foo")`, but in pipelines they run asynchronously when you execute `pipe.Exec(ctx)`

**Comment by LanceEa** [comment]
Thanks for the quick response, that was exactly what I was looking for 😄.
